python/nav/models/sql/changes/sc.05.02.0001.sql

```diff
@@ -0,0 +1,18 @@
+-- Rewrite severity values of existing NAV alert history
+
+-- This function will map any severity value in the interval 0-100 to the new 5-1 interval. The current schema
+-- allows any integer value in the field, so we map any existing values outside the 0-100 interval to either end of the
+-- interval before conversion.
+CREATE FUNCTION pg_temp.severitymap(numeric) RETURNS numeric AS
+$$ SELECT ROUND(((1.0-(GREATEST(0, LEAST($1, 100.0)) / 100.0)) * 4) + 1)  $$
```

Comment:
Nice formula! Works as expected.
Nitpick: I'd still suggest using a conditional expression for the sake of readability.

```
$$ SELECT CASE  WHEN $1 > 80 THEN 1  WHEN $1 > 60 THEN 2  WHEN $1 > 40 THEN 3  WHEN $1 > 20 THEN 4  ELSE 5 END $$
```
Values (80, 60, …) can be adjusted according to preference.

Reply:
I could add LaTeX formatting to the formula to make it more readable 🙈 

In all seriousness, the old values don't matter that much, since no-one ever paid attention to them. 99% of them were `50`, and the rest were `0` or `1`. I could have set all values to 3, and no-one would have cared. In the end, this migration script will disappear once this NAV version becomes so obsolete that we no longer will support migrating from it, so I'm not going to spend much more time on this bit.

I'd rather spend that time writing the issue and the implementation of the now-needed feature: Actually display the severity values, anywhere in the NAV UI, since they are currently invisible to the end user 😆